### PR TITLE
feat(GRO-644): Add sign up via Google to Force

### PR DESCRIPTION
**Pull request by jacobherrington**
This PR allows people to create an account with Google via Force (which we already do in Eigen).

There will be a follow-up task to update the sign-up and sign-in visuals after this in [GRO-639](https://artsyproduct.atlassian.net/browse/GRO-639).

<details>
<summary>Signing up with a Google account</summary>
<a href="https://www.loom.com/share/baf63346515d4dcd8533448f9ac9790d">
    <p>Watch Video</p>
    
</a>
</details>

Thanks, @damassi and @jonallured, for helping me get this done!

JIRA link: https://artsyproduct.atlassian.net/browse/GRO-644

### CODE REVIEW

File: src/dev.js [hunk 1]
```diff
@@ -110,6 +110,7 @@ app.use((req, res, next) => {
 mountAndReload(path.resolve("src/desktop"), {
   watchModules: [
     path.resolve(process.cwd(), "src/v2"),
+    path.resolve(process.cwd(), "src"),
```
Comment (jacobherrington) [inline src/dev.js:113]: @damassi should this stay in the PR?

File: src/lib/passport/test/passport/callbacks.js [hunk 2]
```diff
@@ -1,107 +1,146 @@
...
-const sinon = require('sinon');
-const cbs = rewire('../../lib/passport/callbacks');
+const Backbone = require("backbone")
+const rewire = require("rewire")
+const sinon = require("sinon")
+const cbs = rewire("../../lib/passport/callbacks")
 
-describe('passport callbacks', function() {
+/* eslint-disable jest/expect-expect, jest/no-done-callback */
+// TODO: Refactor artsy-passport code and take care of these warnings!
```
Comment (jacobherrington) [inline src/lib/passport/test/passport/callbacks.js:7]: It seems out of scope to refactor these tests in this PR.

File: src/v2/Apps/Authentication/Components/ModalContainer.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import Cookies from "cookies-js"
 import { Component } from "react"
-import { data as sd } from "sharify"
+import { data as sd } from "sharify" // eslint-disable-line no-restricted-imports
```
Comment (jacobherrington) [inline src/v2/Apps/Authentication/Components/ModalContainer.tsx:3]: It seems out of scope to refactor this component to stop using sharify.

File: src/v2/Components/Authentication/Views/SignUpForm.tsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ import { Formik, FormikProps } from "formik"
 import { Component } from "react"
 import * as React from "react"
 import { recaptcha } from "v2/Utils/recaptcha"
-import { data as sd } from "sharify"
+import { data as sd } from "sharify" // eslint-disable-line no-restricted-imports
```
Comment (jacobherrington) [inline src/v2/Components/Authentication/Views/SignUpForm.tsx:14]: Same as above: It seems out of scope to refactor this component to stop using sharify.

**Review by jacobherrington — COMMENTED** [review]
@damassi would love your feedback on these changes! I want to be sure that I didn't leave anything in from setting up Force to hot-reload when we were still using Yarn to try and do local development with artsy-passport.

File: src/dev.js (hunk 1 above)
Comment (damassi) [reply src/dev.js:113]: Nope! can clear it out

File: src/lib/passport/lib/app/index.js [hunk 5]
```diff
@@ -4,64 +4,80 @@
 // throwing edge case errors that our API returns, and more.
 //
 
-const express = require('express');
-const csrf = require('csurf');
-const passport = require('passport');
-const app = express();
-const opts = require('../options');
+const express = require("express")
```
Comment (damassi) [inline src/lib/passport/lib/app/index.js:7]: Ooh, @jacobherrington - if it is not too late (and its totally fine otherwise) can we open a PR that runs Prettier over `src/lib/passport`
```
yarn prettier src/lib/passport/* --write
```
and then rebase into this PR? With these cosmetic changes its making it difficult to see what the actual changes are.

Comment (damassi) [reply src/lib/passport/lib/app/index.js:7]: (Or actually, lemme do that real quick since its my bad! one sec...)

**Comment by jacobherrington** [comment]
Before this is merged, we'll need to configure staging and prod in GCP and set some keys, but I can make sure that happens when this goes to staging.

File: src/lib/passport/lib/app/index.js (hunk 5 above)
Comment (damassi) [reply src/lib/passport/lib/app/index.js:7]: done and merged: https://github.com/artsy/force/pull/9290

Comment (jacobherrington) [reply src/lib/passport/lib/app/index.js:7]: Cool! I rebased and fixed the merge conflicts.

File: src/lib/passport/test/passport/callbacks.js (hunk 2 above)
Comment (damassi) [reply src/lib/passport/test/passport/callbacks.js:7]: Totally! don't worry about it.

File: src/v2/Apps/Authentication/Components/ModalContainer.tsx (hunk 3 above)
Comment (damassi) [reply src/v2/Apps/Authentication/Components/ModalContainer.tsx:3]: Mind adding a ticket in the web practice board [here](https://artsyproduct.atlassian.net/jira/software/projects/WP/boards/97) to remove this flag? Can future friday the fix.

File: src/v2/Components/Authentication/Views/SignUpForm.tsx (hunk 4 above)
Comment (damassi) [reply src/v2/Components/Authentication/Views/SignUpForm.tsx:14]: (This can also go in this ticket here https://github.com/artsy/force/pull/9143/files#r794756516)

File: src/v2/server.ts [hunk 6]
```diff
@@ -23,6 +42,39 @@ if (!NOVO_MANIFEST) {
...
+      "id",
+      "lab_features",
+      "name",
+      "paddle_number",
+      "phone",
+      "roles",
+      "type",
+    ],
+  })
+)
```
Comment (damassi) [inline src/v2/server.ts:76]: We should remove this code as its related to the hot reloading fix:
```
git checkout main -- src/v2/server.ts
```

**Review by damassi — DISMISSED** [review]
Only a couple minor fixes, but otherwise LGTM! Thanks for pushing through all the confusion on this 👍

**Review by damassi — APPROVED** [review]

**Comment by jacobherrington** [comment]
Keys are set for staging, I'll test it there and then set up the keys for production!

**Comment by jonallured** [comment]
So excited for you that this is merged!! 🏆
